### ISSUE TITLE
Latest version of tinyspy (3.0.1) causes a `Cannot redefine property: mock` when trying to spyOn an object multiple times

### ISSUE THREAD

**Issue by vladfrangu** [vitest-dev/vitest#6465]
### Describe the bug

Renovate caught this issue in our CI (https://github.com/apify/apify-cli/pull/636). It seems like the latest version of tinyspy (3.0.1) causes some issues when trying to spy on and mock an object's method multiple times. This issue doesn't repro on tinyspy 3.0.0 (and somehow I was also unable to reproduce this with tinyspy directly...)

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-uuvjoy

### System Info

```shell
System:
    OS: macOS 15.0
    CPU: (10) arm64 Apple M1 Max
    Memory: 150.05 MB / 32.00 GB
    Shell: 5.9 - /opt/homebrew/bin/zsh
  Binaries:
    Node: 20.17.0 - ~/.volta/tools/image/node/20.17.0/bin/node
    Yarn: 4.4.1 - ~/.volta/tools/image/yarn/4.4.1/bin/yarn
    npm: 10.8.2 - ~/.volta/tools/image/node/20.17.0/bin/npm
    bun: 1.1.26 - ~/.bun/bin/bun
  Browsers:
    Chrome: 128.0.6613.120
    Safari: 18.0
  npmPackages:
    vitest: ^2.0.5 => 2.0.5
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by vladkrasn** [comment]
Yup, getting this in my ci too. Typescript's intellisense also doesn't pickup import

**Comment by sheremet-va** [comment]
Reverted the change in `tinyspy@3.0.2`, so renovate shouldn't pick it up again. The change instead will be published as 4.0.0.
